src/lib/layer-manager.js

```diff
@@ -114,9 +90,14 @@ export default class LayerManager {
   }
 
   updateLayers({newLayers}) {
-    /* eslint-disable */
-    assert(this.context.viewport,
-      'LayerManager.updateLayers: viewport not set');
+    // TODO - something is generating state updates that cause rerender of the same
+    if (newLayers === this.lastRenderedLayers) {
```

Comment:
I'm a bit confused over this newly added `lastRenderedLayers`... we probably should have a doc explaining the relations between `this.prevLayers`, `this.lastRenderedLayers`, `this.layers`, etc... (and also additional variables like `generatedLayers`, `oldLayers`, `newLayers`)